engine/runtime-parser/src/main/scala/org/enso/compiler/core/ir/LazyDiagnosticStorage.scala

```diff
@@ -0,0 +1,32 @@
+package org.enso.compiler.core.ir
+
+import org.enso.compiler.core.IR
+
+trait LazyDiagnosticStorage { self: IR =>
+
+  private[this] var _diagnostics: DiagnosticStorage = _
+
+  protected def diagnostics_=(diagnostics: DiagnosticStorage): Unit = {
```

Comment:
Who's calling setter (this is setter, right)? There was no setter before, so I don't think we need a method for it now.

Reply:
We need a setter when creating an IR node and want to pass some existing diagnostic storage.